File: backend/transaction-service/src/main/java/com/gl/transaction_service/entity/Transaction.java
package com.gl.transaction_service.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long transactionId;


    // -------------------------
    // BOOKING / RESOURCE
    // -------------------------

    private Long bookingId;

    private Long resourceId;


    // -------------------------
    // USERS
    // -------------------------

    private Long renterId;

    private Long ownerId;


    // -------------------------
    // RENT DETAILS
    // -------------------------

    private Double rentPerDay;

    private Integer rentalDays;

    private Double totalRent;

    private Double securityDeposit;


    // -------------------------
    // PAYMENT
    // -------------------------



    @Enumerated(EnumType.STRING)
    private PaymentStatus paymentStatus;


    // -------------------------
    // DATES
    // -------------------------

    private LocalDate bookingDate;

    private LocalDate expectedReturnDate;

    private LocalDateTime createdAt;


    // -------------------------
    // TRANSACTION STATUS
    // -------------------------

    @Enumerated(EnumType.STRING)
    private TransactionStatus status;


    // -------------------------
    // CHECKBOXES
    // -------------------------

    private Boolean rentPaid;

    private Boolean securityDepositPaid;

    private Boolean resourceCollected;

    private Boolean resourceReturned;

    private Boolean securityDepositReturned;


    // -------------------------
    // CREATED TIME
    // -------------------------

    @PrePersist
    public void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/transaction.ts
export type PaymentStatus =
  | "PENDING"
  | "PAID"
  | "REFUNDED";

export type TransactionStatus =
  | "ACTIVE"
  | "COMPLETED";

export interface Transaction {

  transactionId: number;

  bookingId: number;

  resourceId: number;

  renterId: number;

  ownerId: number;

  rentPerDay: number;

  rentalDays: number;

  totalRent: number;

  securityDeposit: number;

  bookingDate: string;

  expectedReturnDate: string;

  paymentStatus: PaymentStatus;

  status: TransactionStatus;

  rentPaid: boolean;

  securityDepositPaid: boolean;

  resourceCollected: boolean;

  resourceReturned: boolean;

  securityDepositReturned: boolean;

  createdAt: string;

  resourceName: string;
  ownerName: string;
  borrowerName: string;
}